standalone-metastore/metastore-server/src/main/sql/hive/hive-schema-4.1.0.hive.sql

```diff
@@ -1592,6 +1592,57 @@ WHERE
     AND
     B.PARAM_KEY LIKE 'repl_metrics%';
 
+CREATE EXTERNAL TABLE IF NOT EXISTS `PROTO_HIVE_QUERY_DATA`
+PARTITIONED BY (
+  `date` string
+)
+ROW FORMAT SERDE 'org.apache.hadoop.hive.ql.io.protobuf.ProtobufMessageSerDe'
+STORED AS INPUTFORMAT 'org.apache.hadoop.hive.ql.io.protobuf.ProtobufMessageInputFormat'
+OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'
+LOCATION '_REPLACE_WITH_QUERY_DATA_LOCATION_'
+TBLPROPERTIES (
+  'proto.class'='org.apache.hadoop.hive.ql.hooks.proto.HiveHookEvents$HiveHookEventProto',
+  'proto.maptypes'='org.apache.hadoop.hive.ql.hooks.proto.MapFieldEntry'
+);
```

Comment:
I pulled this branch, ran schematool, and am checking the behavior. Any partitions don't appear for me unless I add partitions by hand. Is this an expected behavior?

```
schematool -metaDbType derby -dbType hive -initSchema -url jdbc:hive2://hive-hiveserver2:10000/default -driver org.apache.hive.jdbc.HiveDriver
```

```
0: jdbc:hive2://hive-hiveserver2:10000/defaul> show partitions sys.PROTO_HIVE_QUERY_DATA;
...
+------------+
| partition  |
+------------+
+------------+
```

Manual ADD PARTITION makes my data visible.

```
0: jdbc:hive2://hive-hiveserver2:10000/defaul> ALTER TABLE sys.PROTO_HIVE_QUERY_DATA ADD PARTITION (`date`='2024-06-23') LOCATION 'hdfs:///tmp/hive-proto/date=2024-06-23';
...
0: jdbc:hive2://hive-hiveserver2:10000/defaul> select count(*) from sys.PROTO_HIVE_QUERY_DATA;
...
+------+
| _c0  |
+------+
| 83   |
+------+
...
```

Reply:
Yes, this is expected behaviour and the partitions have to be added manually, or MSCK REPAIR TABLE could be executed. This PR is for exposing these protologging tables. However, the partitions could be added automatically by enabling metastore housekeeping threads and adding PartitionManagementTask to run in the background. In case, protologging is not enabled, I've used a default location to use for these tables as fallback.

Reply:
In the PR commit build I am getting this error, but I have not changed anything in TestMiniLlapLocalCliDriver. My changes are only related to schema and I'm unable to understand as to why this specific test is failing. Could you please help me with fixing the build? Thank you.

> ./itests/qtest/target/surefire-reports/TEST-org.apache.hadoop.hive.cli.split23.TestMiniLlapLocalCliDriver.xml:59905.176: internal error: Huge input lookup

Reply:
Thanks. I am neither a user of `hive.sys` nor a Hive committer, so I am not sure what definitions are best for the community. I think the committers will make the final decision about whether we should have those tables here or how we define the schema.